clusterloader2/docs/test_suite_design.md

```diff
@@ -0,0 +1,46 @@
+# Cluster loader - TestSuite
+
+## Background
+
+Since 08/2018 cluster loader become main tool for kubernetes e2e performance tests.
+It's is a very convenient tool for running single test case, but not for running
+test bundles. Currently it is impossible to run the same test
+with two different parameters sets in a single job run.
+This forces developers to create separate job configs for
+almost each tests.
+
+## TestSuite
+
+TestSuite is a test bundle.
+Overrides is a set of override file paths. Overrides provided on the TestSuite level will be applied
+to all of the test, unless they are overidden on the test level.
+
+```
+type TestSuite struct {
+	Overrides []string
+	Tests     []Test
+}
+
+```
+
+## Test
+
+Test defines a single test scenario. Config defines path to the config file
+that satisfy already existing cluster loader API. Overrides defines what
+override files should be applied to the config file.
+To enable using same test config in one test suite,
+the Identifier field was introduced. This indetifier will be used to match
+run results to the corresponding test.
+
+```
+type Test struct {
+	Overrides  []string
```

Comment:
Would be good to add a word or two about override priorities. 
If we have two level of overrides it's easy to imagine a situation where I override something at the suite level and test level at the same time. E.g. I set LOAD_TEST_THROUGHPUT to 20 at suite level and to 50 at some test level. I assume in such case the test level override will take precedence over suite level, right?

Reply:
I added a comment that I vote for removing the "top-level" overrides.

Reply:
I've removed top level overrides.